scripts/rollup/packaging.js

```diff
@@ -165,15 +166,48 @@ function copyNodePackageTemplate(packageName) {
     // We already created this package (e.g. due to another entry point).
     return Promise.resolve();
   }
-  // TODO: verify that all copied files are either in the "files"
-  // whitelist or implicitly published by npm.
-  return asyncCopyTo(npmFrom, to).then(() =>
-    Promise.all([
+  fs.mkdirSync(to);
+  const packageJson = resolve(`${from}/package.json`);
+  const whitelistedFiles = fs.existsSync(packageJson)
+    ? require(packageJson).files
+    : null;
+  const npmRootFiles = fs.readdirSync(npmFrom);
+  // Get all entry points in the package root
+  // Exceptions: *.fb.js
+  const packageRootEntries = glob.sync('!(*.fb).js', {
+    cwd: from,
+  });
+  packageRootEntries.forEach(entry => {
+    // Terminate the build if any entry point in the package root does not have an equivalent in ./npm.
+    if (!npmRootFiles.includes(entry)) {
+      console.error(
+        `Entry point ${entry} in package ${packageName} does not have an equivalent in ./npm`
+      );
+      process.exit(1);
+    }
+  });
+  if (whitelistedFiles && whitelistedFiles.length > 0) {
+    let asyncCopyProxies = [];
+    asyncCopyProxies = npmRootFiles.reduce((proxies, file) => {
+      if (
```

Comment:
A comment here would help understand what this does and why

Reply:
Comments added